lib/internal/modules/esm/translators.js

```diff
@@ -140,35 +142,103 @@ function enrichCJSError(err, content, filename) {
   }
 }
 
-// Strategy for loading a node-style CommonJS module
-const isWindows = process.platform === 'win32';
-translators.set('commonjs', async function commonjsStrategy(url, source,
-                                                            isMain) {
+/**
+ * Loads a CommonJS module via the ESM Loader sync CommonJS translator.
+ * This translator creates its own version of the `require` function passed into CommonJS modules.
+ * Any monkey patches applied to the CommonJS Loader will not affect this module.
+ * Any `require` calls in this module will load all children in the same way.
+ */
+function loadCJSModule(module, source, url, filename) {
+  let compiledWrapper;
+  try {
+    compiledWrapper = internalCompileFunction(source, [
+      'exports',
+      'require',
+      'module',
+      '__filename',
+      '__dirname',
+    ], {
+      filename,
+      importModuleDynamically(specifier, _, importAssertions) {
+        return asyncESM.esmLoader.import(specifier, url, importAssertions);
+      },
+    }).function;
+  } catch (err) {
+    enrichCJSError(err, source, url);
+    throw err;
+  }
+
+  const __dirname = dirname(filename);
+  // eslint-disable-next-line func-name-matching,func-style
+  const requireFn = function require(specifier) {
+    let importAssertions = kEmptyObject;
+    if (!StringPrototypeStartsWith(specifier, 'node:')) {
+      // TODO: do not depend on the monkey-patchable CJS loader here.
+      const path = CJSModule._resolveFilename(specifier, module);
+      if (specifier !== path) {
+        switch (extname(path)) {
+          case '.json':
+            importAssertions = { __proto__: null, type: 'json' };
+            break;
+          case '.node':
+            return CJSModule._load(specifier, module);
+          default:
```

Comment:
nit: this looks like a mistake

```suggestion
          default: break;
```

Reply:
I'd argue that `default: break` looks even more like a mistake 😅 I'm going to add a comment instead.